header-only/awkward/utils.h

```diff
@@ -72,10 +84,18 @@ namespace awkward {
   const std::string
   type_to_name<bool>() {
     // This takes precedence over the unspecialized template, and therefore any
-    // 8-bit data that is not named bool will be mapped to "int8" or "uint8".
+    // 8-bit data that is not named char will be mapped to "int8" or "uint8".
     return "bool";
   }
```

Comment:
I know that this is just a comment, but this function specifically catches the types that the compiler recognizes as `bool` at compile-time. Maybe that includes `char`; I actually don't know why the compiler considers some equivalent types "the same" and others "not the same" when matching a template instantiation. (In fact, I just hope it's not compiler-dependent!)

Did you miss the word "not" here? "8-bit data that is not named bool" is not handled here (returning the string `"bool"`); it's handled in the generic version of this function.

Reply:
oh, wrong comment :-)

Reply:
thanks for spotting it - I've corrected the typo. Thanks